### PR TITLE
Fix MockPlugin behavior in tests

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - Fix MockPlugin behavior in tests

### Tickets:
 - CVS-134410

### CODE REVIEW

File: src/tests/test_utils/unit_test_utils/mocks/mock_engine/mock_plugin.cpp [hunk 1]
```diff
@@ -190,3 +190,8 @@ OPENVINO_PLUGIN_API void InjectPlugin(ov::IPlugin* target) {
     std::lock_guard<std::mutex> lock(targets_mutex);
     targets.push(std::make_shared<MockInternalPlugin>(target));
 }
+
+OPENVINO_PLUGIN_API void ClearTargets() {
+    decltype(targets) empty;
+    std::swap(targets, empty);
```
Comment (praasz) [inline src/tests/test_utils/unit_test_utils/mocks/mock_engine/mock_plugin.cpp:196]: Why is swap required?
```suggestion
    targets = {}
```
It should release all items from queue. Should lock be applied as `targets` is global object and when modified is protected by lock in `create_plugin_engine` function?

File: src/inference/tests/functional/ov_register_plugin_test.cpp [hunk 2]
```diff
@@ -36,6 +37,13 @@ inline void mockPlugin(ov::Core& core, std::shared_ptr<ov::IPlugin>& plugin, std
     injectProxyEngine(plugin.get());
 }
 
+void clearMockPlugin(const std::shared_ptr<void>& m_so) {
+    std::string libraryPath = ov::test::utils::get_mock_engine_path();
```
Comment (praasz) [inline src/inference/tests/functional/ov_register_plugin_test.cpp:41]:
```suggestion
```
Is not used.

File: src/tests/test_utils/unit_test_utils/mocks/mock_engine/mock_plugin.cpp (hunk 1 above)
Comment (olpipi) [reply src/tests/test_utils/unit_test_utils/mocks/mock_engine/mock_plugin.cpp]: done

File: src/inference/tests/functional/ov_register_plugin_test.cpp (hunk 2 above)
Comment (olpipi) [reply src/inference/tests/functional/ov_register_plugin_test.cpp]: done

**Review by praasz — APPROVED** [review]

**Pull request merged**